client/client.go

```diff
@@ -0,0 +1,604 @@
+
+	servicePath := filepath.Join(systemdRootPath, "ccloudvm.service")
+	_ = os.Remove(servicePath)
+
+	socketPath := filepath.Join(systemdRootPath, "ccloudvm.socket")
+	_ = os.Remove(socketPath)
+	return nil
+}
+
+func dialHTTP(ctx context.Context, socketPath string) (client *rpc.Client, err error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	d := &net.Dialer{}
+	conn, err := d.DialContext(timeoutCtx, "unix", socketPath)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if client == nil {
+			_ = conn.Close()
+		}
+	}()
+	connectString := fmt.Sprintf("CONNECT %s HTTP/1.0\n\n", rpc.DefaultRPCPath)
+	_, err = io.WriteString(conn, connectString)
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := http.ReadResponse(bufio.NewReader(conn), &http.Request{
+		Method: http.MethodConnect,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if r.Status != "200 Connected to Go RPC" {
+		return nil, errors.Errorf("Unexpected response (%s) from RPC server", r.Status)
+	}
+
+	return rpc.NewClient(conn), nil
+}
+
+func issueCommand(ctx context.Context, call func(*rpc.Client) (int, error),
+	result func(*rpc.Client, int) error) error {
+	home := os.Getenv("HOME")
+	if home == "" {
+		return errors.New("HOME is not defined")
+	}
+
+	socketPath := filepath.Join(home, ".ccloudvm/socket")
+	client, err := dialHTTP(ctx, socketPath)
+	if err != nil {
+		err2 := exec.Command("systemctl", "--user", "restart", "ccloudvm.socket").Run()
+		if err2 != nil {
+			return errors.Wrap(err, "Unable to communicate with server. Try running 'ccloudvm setup'")
+		}
+		client, err = rpc.DialHTTP("unix", socketPath)
+		if err != nil {
+			return errors.Wrap(err, "Unable to communicate with server")
+		}
+	}
+
+	defer func() {
+		_ = client.Close()
+	}()
+
+	id, err := call(client)
+	if err != nil {
+		return err
+	}
+
+	ch := make(chan error)
+	go func() {
+		ch <- result(client, id)
+	}()
+
+	reply := struct{}{}
+
+	select {
+	case err = <-ch:
+		return err
+	case <-ctx.Done():
+		_ = client.Call("ServerAPI.Cancel", id, &reply)
+		return <-ch
+	}
+}
+
+func getProxy(upper, lower string) (string, error) {
+	proxy := os.Getenv(upper)
+	if proxy == "" {
+		proxy = os.Getenv(lower)
+	}
+
+	if proxy == "" {
+		return "", nil
+	}
+
+	if proxy[len(proxy)-1] == '/' {
+		proxy = proxy[:len(proxy)-1]
+	}
+
+	proxyURL, err := url.Parse(proxy)
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to parse %s", proxy)
+	}
+	return proxyURL.String(), nil
+}
+
+// Create sets up the VM
+func Create(ctx context.Context, instanceName, workloadName string, debug bool, update bool, customSpec *types.VMSpec) error {
+	HTTPProxy, err := getProxy("HTTP_PROXY", "http_proxy")
+	if err != nil {
+		return err
+	}
+
+	HTTPSProxy, err := getProxy("HTTPS_PROXY", "https_proxy")
+	if err != nil {
+		return err
+	}
+
+	if HTTPSProxy != "" {
+		u, _ := url.Parse(HTTPSProxy)
+		u.Scheme = "http"
+		HTTPSProxy = u.String()
+	}
+
+	noProxy := os.Getenv("no_proxy")
```

Comment:
and NO_PROXY ?

Reply:
Good point.  I'll add this.  First I need to check whether the code handles NO_PROXY in general.  If not, I'll enter a bug and fix this in a separate PR.